etcdadm/etcdadm

```diff
@@ -0,0 +1,1559 @@
+# Gotcha: ETCDCTL_ENDPOINT seems to be preferred over --peers flag, which results in `etcdctl --peers` to be useless
+unset ETCDCTL_ENDPOINT
+
+export | grep ETCDCTL
+
+ETCD_WORK_DIR=${ETCD_WORK_DIR:-$(pwd)/work}
+
+if [ "${DEBUG:-}" == "yes" ]; then
+  set -vx
+fi
+
+_info() {
+  echo "$0: info: ${FUNCNAME[1]}: $*" >&2
+}
+
+_error() {
+  echo "$0: error: ${FUNCNAME[1]}: $*" >&2
+}
+
+_panic() {
+  echo "$0: panic: ${FUNCNAME[1]}: $*" >&2
+  exit 1
+}
+
+_array_join() {
+  local r=""
+  for t in "${@}"; do
+    if [ "$r" != "" ]; then
+      r="$r \"$t\""
+    else
+      r="\"$t\""
+    fi
+  done
+  echo "$r"
+}
+
+_current_time() {
+  date +%s
+}
+
+_run_as_root() {
+  local status
+  if [[ "$EUID" > 0 ]]; then
+    echo running command as root: "$(_array_join "$@")"
+    _array_join "${@}" | sudo bash
+  else
+    _array_join "${@}" | bash
+  fi
+  status=$?
+  return "$status"
+}
+
+config_awscli_docker_image() {
+  echo "${ETCDADM_AWSCLI_DOCKER_IMAGE:-quay.io/coreos/awscli}"
+}
+
+awscli_rkt_image() {
+  echo "docker://$(config_awscli_docker_image)"
+}
+
+config_aws_region() {
+  echo "${AWS_DEFAULT_REGION:-ap-northeast-1}"
+}
+
+config_aws_access_key_id() {
+  echo "${AWS_ACCESS_KEY_ID}"
+}
+
+config_aws_secret_access_key() {
+  echo "${AWS_SECRET_ACCESS_KEY}"
+}
+
+config_etcd_initial_cluster() {
+  echo "${ETCD_INITIAL_CLUSTER}"
+}
+
+config_etcd_endpoints() {
+  echo "${ETCD_ENDPOINTS}"
+}
+
+config_etcd_version() {
+  echo "${ETCD_VERSION:-3.1.3}"
+}
+
+config_member_count() {
+  echo "${ETCDADM_MEMBER_COUNT:-3}"
+}
+
+config_member_index() {
+  echo "${ETCDADM_MEMBER_INDEX}"
+}
+
+config_member_systemd_unit_name() {
+  echo "${ETCDADM_MEMBER_SYSTEMD_UNIT_NAME:-$(config_member_systemd_service_name).service}"
+}
+
+config_member_systemd_service_name() {
+  echo "${ETCDADM_MEMBER_SYSTEMD_SERVICE_NAME:-etcd-member-$(config_member_index)}"
+}
+
+config_cluster_snapshots_s3_uri() {
+  echo "${ETCDADM_CLUSTER_SNAPSHOTS_S3_URI:?missing required env}"
+}
+
+config_state_dir() {
+  echo "${ETCDADM_STATE_FILES_DIR:-/var/run/coreos/$(member_name)-state}"
+}
+
+cluster_etcd_aci_url() {
+  local v
+  v=v$(config_etcd_version)
+  echo "https://github.com/coreos/etcd/releases/download/$v/etcd-$v-linux-amd64.aci"
+}
+
+cluster_member_indices() {
+  i=0
+  until [ "$i" == "$(config_member_count)" ]; do
+    echo $i
+    i=$((i + 1))
+  done
+}
+
+cluster_is_healthy() {
+  ! cluster_is_unhealthy
+}
+
+# i.e. cluster_quorum_may_have_been_lost. The lose may or may not be permanent.
+# We don't have way to determine whether it is permanent or transient?
+cluster_is_unhealthy() {
+  local healthy
+  local quorum
+  healthy=$(cluster_num_healthy_members)
+  quorum=$(cluster_majority)
+  _info "quorum=$quorum healthy=$healthy"
+  if (( healthy < quorum )); then
+    _info 'cluster is unhealthy'
+    return 0
+  fi
+  _info 'cluster is healthy'
+  return 1
+}
+
+cluster_majority() {
```

Comment:
It was written as so to prevent successful `echo` even when `config_member_count` is failed.
However, making it just a variable would be better as I have `set -o nounset` now 👍